### PR TITLE
Update go_genrule to work with recent rules_go changes

### PR DESCRIPTION
**Pull request by ixdy**
The upstream Bazel go rules were updated recently to limit the public API and to prepare for Bazel toolchain work. This PR adapts these changes and kicks the can down the road a bit. (Things may very well break again with further Bazel toolchain changes.)

### CODE REVIEW

File: defs/go.bzl [hunk 1]
```diff
@@ -70,7 +66,7 @@ def _go_genrule_impl(ctx):
   ctx.action(
       inputs = list(all_srcs) + resolved_inputs,
       outputs = ctx.outputs.outs,
-      env = ctx.configuration.default_shell_env,
+      env = ctx.configuration.default_shell_env + ctx.attr._go_toolchain.env,
```
Comment (ixdy) [inline defs/go.bzl]: stuff like this came from https://github.com/bazelbuild/rules_go/blob/master/go/private/go_toolchain.bzl#L51-L71

**Comment by spxtr** [comment]
How did you test that this works?

**Comment by ixdy** [comment]
I've tested it locally and it seems to do the right thing (i.e. it successfully builds). I'm not sure how to test that the output is necessarily correct.

See also https://github.com/kubernetes/kubernetes/pull/47004.

**Comment by spxtr** [comment]
Looks fine to me, @mikedanese should give the green mark.

**Pull request merged**